src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if ((key == keys[i]) || key != null && key.equals(keys[i])) {
+                values[i] = value;
+                return;
+            }
+        }
+
+        if (size < MAX_SIZE) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            throw new IllegalStateException("Storage is full");
+        }
```

Comment:
better to have less identation as possible
```suggestion
        if (size >= MAX_SIZE) {
            throw new IllegalStateException("Storage is full");
        } 
        keys[size] = key;
        values[size] = value;
        size++;
```